core/src/types.rs

```diff
@@ -383,6 +418,7 @@ pub struct SyncClientConfig {
 	pub confidence: f64,
 	pub disable_rpc: bool,
 	pub app_id: Option<u32>,
+	pub network_mode: NetworkMode,
```

Comment:
We can remove `disable_rpc` entirely and use equality directly in the code, since we are storing network_mode in the configuration.